src/collection.js

```diff
@@ -322,7 +323,38 @@ export default class Collection {
   listRecords(options={}) {
     const path = endpoint("record", this.bucket.name, this.name);
     const reqOptions = this._collOptions(options);
-    return this.client.paginatedList(path, options, reqOptions);
+    if (options.hasOwnProperty("at")) {
+      return this._getSnapshot(options.at);
+    } else {
+      return this.client.paginatedList(path, options, reqOptions);
+    }
+  }
+
+  _getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    return this.bucket.listHistory({
+      filters: {
+        resource_name: "record",
+        collection_id: this.name,
+        "max_target.data.last_modified": String(at),
+      }
+    })
+      .then(({data: changes}) => {
+        const seenIds = new Set();
+        let snapshot = [];
+        for (const {target: {data: record}} of changes) {
+          if (record.deleted) {
+            seenIds.add(record.id);
+            snapshot = snapshot.filter(r => r.id !== record.id);
+          } else if (!seenIds.has(record.id)) {
+            seenIds.add(record.id);
+            snapshot.push(record);
+          }
+        }
+        return snapshot.sort((a, b) => b.last_modified - a.last_modified);
```

Comment:
That's a good idea, I'm just slightly afraid that would open the door to supporting all other options. I liked the opinionated simplistic approach here, but we may want to think about it further.